x-pack/test/apm_api_integration/tests/settings/anomaly_detection/no_access_user.spec.ts

```diff
@@ -6,38 +6,55 @@
  */
 
 import expect from '@kbn/expect';
+import { ApmApiError } from '../../../../../../node_modules/x-pack/test/apm_api_integration/common/apm_api_supertest';
 import { FtrProviderContext } from '../../../common/ftr_provider_context';
 
 export default function apiTest({ getService }: FtrProviderContext) {
   const registry = getService('registry');
-  const noAccessUser = getService('legacySupertestAsNoAccessUser');
+  const apmApiClient = getService('apmApiClient');
 
   function getJobs() {
-    return noAccessUser.get(`/internal/apm/settings/anomaly-detection/jobs`).set('kbn-xsrf', 'foo');
+    return apmApiClient.noAccessUser({
+      endpoint: 'GET /internal/apm/settings/anomaly-detection/jobs',
+    });
   }
 
   function createJobs(environments: string[]) {
-    return noAccessUser
-      .post(`/internal/apm/settings/anomaly-detection/jobs`)
-      .send({ environments })
-      .set('kbn-xsrf', 'foo');
+    return apmApiClient.noAccessUser({
+      endpoint: 'POST /internal/apm/settings/anomaly-detection/jobs',
+      params: {
+        body: {
+          environments,
+        },
+      },
+    });
   }
 
   registry.when('ML jobs', { config: 'trial', archives: [] }, () => {
     describe('when user does not have read access to ML', () => {
       describe('when calling the endpoint for listing jobs', () => {
         it('returns an error because the user does not have access', async () => {
-          const { body } = await getJobs();
-          expect(body.statusCode).to.be(403);
-          expect(body.error).to.be('Forbidden');
+          try {
+            await getJobs();
```

Comment:
If there is no error thrown, do we get a failed test in this case as there are no assertions? You could also use something like [this](https://github.com/elastic/kibana/blob/b0432083a61c5df9a8695c6308e11e90a228ed01/x-pack/test/apm_api_integration/tests/settings/custom_link.spec.ts#L184) and even convert it to a helper. The applies to the other spec as well

Reply:
Very nice suggestion. Now the code looks ❤️